backends/p4tools/testgen/targets/ebpf/backend/stf/stf.h

```diff
@@ -0,0 +1,79 @@
+#ifndef BACKENDS_P4TOOLS_TESTGEN_TARGETS_EBPF_BACKEND_STF_STF_H_
+#define BACKENDS_P4TOOLS_TESTGEN_TARGETS_EBPF_BACKEND_STF_STF_H_
+
+#include <cstddef>
+#include <fstream>
+#include <map>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <boost/optional/optional.hpp>
+#include <inja/inja.hpp>
+
+/// Inja
+#include "ir/ir.h"
+#include "lib/cstring.h"
+
+#include "backends/p4tools/testgen/lib/tf.h"
+#include "backends/p4tools/testgen/targets/ebpf/test_spec.h"
+
+namespace P4Tools {
+
+namespace P4Testgen {
+
+namespace EBPF {
+
+/// Extracts information from the @testSpec to emit a STF test case.
+class STF : public TF {
```

Comment:
I repeat the question about what in this class is specific to EBPF. It looks to me like STF should be the same for all backends.

Reply:
There are subtle difference between how the STF tests are processed for each back end. For example, BMv2 does not need [this](https://github.com/p4lang/p4c/pull/3510/files/2fe95c34809d2682684aee35790cdff58e73a0c3#diff-792eb6879e16c8c09289fd83c718ffa32b0baa5ea1235157fa0d13dfde2703afR98) and does support `$` as a postfix to indicate exact packet length matching. Furthermore, BMv2 supports various externs such as clone, action profiles, registers, which are handled in this stf class.